lib/libc/Makefile

```diff
@@ -35,11 +35,13 @@ CFLAGS+=-D_FORTIFY_SOURCE_read=_read
 # To include RCS ID strings from other BSD projects, remove -DNO__RCSID from CFLAGS.
 CFLAGS+=-DNO__SCCSID -DNO__RCSID
 
-.ifdef CHERI_LIB_C18N
-SHLIB=c
-.else
-LIB?=c
+.if ${MACHINE_ABI:Mpurecap} && ${MACHINE_CPUARCH} == "aarch64"
+.if !defined(COMPAT_LIBCOMPAT) || ${COMPAT_LIBCOMPAT} == "64CB"
```

Comment:
Hmm, I thought we altered MACHINE_ABI while COMPAT_LIBCOMPAT was in force?  Note also that "64C" can still happen when building a hybrid world.  I just wonder if you need this COMPAT_LIBCOMPAT check at all?

Reply:
This is because we didn't have a /usr/lib64c/c18n for hybrid worlds IIRC (since it's a separate case to /usr/lib/c18n in etc/mtree and Makefile.inc1). But now it's just in the normal purecap libraries we don't need to disable c18n for hybrid worlds. Though I guess we still had a /usr/lib64cb/c18n there, which is a bit weird.